content/blog/2022/04/securing-kubewarden-policies.md

```diff
@@ -0,0 +1,209 @@
+$ echo $?
+0
+```
+
+
+All the policies from the Kubewarden team are signed in keyless mode by the
+workers of the CI job. In our case, the workers used in the CI job of Github.
+We don't leave certs around, and they are verifiable by third parties.
+
+### Enforcing signature verification for instantiated Kubewarden policies
+
+You can now configure
+[PolicyServers](https://doc.crds.dev/github.com/kubewarden/kubewarden-controller/policies.kubewarden.io/PolicyServer/v1alpha2)
+to enforce that all policies being run will need to be signed. And when
+deploying Kubewarden via Helm charts, you can do it so for the default
+PolicyServer installed by `kubewarden-defaults` chart.
+
+For this, the PolicyServers have a new [`spec.VerificationConfig`](https://doc.crds.dev/github.com/kubewarden/kubewarden-controller/policies.kubewarden.io/PolicyServer/v1alpha2#spec-verificationConfig) argument.
+Here, you can put the name of a ConfigMap containing a "verification config", to
+specify the needed signatures.
+
+You can obtain a default verification config for policies from the Kubewarden
+team with:
+
+```yaml
+$ kwctl scaffold verification-config
+# Default Kubewarden verification config
+#
+# With this config, the only valid policies are those signed by Kubewarden
+# infrastructure.
+#
+# This config can be saved to its default location (for this OS) with:
+#   kwctl scaffold verification-config > /home/youruser/.config/kubewarden/verification-config.yml
+#
+# Providing a config in the default location enables Sigstore verification.
+# See https://docs.kubewarden.io for more Sigstore verification options.
+---
+apiVersion: v1
+allOf:
+  - kind: githubAction
+    owner: kubewarden
+    repo: ~
+    annotations: ~
+anyOf: ~
+```
+
+The verification config format has several niceties, [see its reference docs](TODO point to VerificationConfig reference in docs).
+For example, `kind: githubAction` with `owner` and `repo`, instead of checking the `issuer` and
+`subject` strings blindly. Or `anyOf` a list of signatures, with `anyOf.atLeast` a number
+of them: this allows for accepting at least a specific number of signatures, and
+makes migration between signatures in your cluster easy. It's the little things :).
+
+If you want support for other CIs (such as Gitlab, Jenkins, etc) drop us some
+words!
+
+Once you have crafted your verification config, create your ConfigMap:
+```console
+$ kubectl create configmap my-verification-config \
+  --from-file=verification-config=./my-verification-config.yml \
+  --namespace=kubewarden
+```
+
+And pass it to your PolicyServers in `spec.VerificationConfig`, or if using the
+default PolicyServer from the `kubewarden-defaults` chart, set it there with for example:
+```console
+$ helm upgrade --set policyServer.verificationConfig=my-verification-config \
+  --wait -n kubewarden kubewarden-defaults ./kubewarden-defaults
+```
+
+## Recap
+
+Using `cosign`, `kwctl verify`, and  `kwctl inspect`, we can sign our authored
```

Comment:
we use `cosign` to sign, and `kwctl verify` to verify.

> Using `cosign`, `kwctl verify`, and  `kwctl inspect`, we can sign our authored

In this sentence it looks to me that you need these 3 tools to sign

Reply:
True! done in https://github.com/kubewarden/kubewarden.io/pull/82/commits/29edb820d16ba573bdd80b698087c17b69102e25